### ISSUE TITLE
Linux: Option to Set Urgent Hint

### ISSUE THREAD

**Issue by zepalmer** [mattermost/desktop#122]
I am using the Mattermost desktop client (v1.1.1) under XFCE4 on Debian Stretch (testing).  It would be convenient to have an option in the Mattermost client that would set the X window manager's "urgent" hint on the Mattermost window upon receiving a message.  In my desktop environment, this causes the window's task bar button to be highlighted.

I use Pidgin for some of my messaging and a Pidgin plugin, Message Notification 2.10.11, supports this capability.

**Comment by yuya-oc** [comment]
Excuse me, I don't know about "urgent hint", could you tell me some reference or screenshots?

**Comment by zepalmer** [comment]
Here's a picture of the Pidgin plugin menu showing the plugin:

![screenshot_2016-04-29_12-09-49](https://cloud.githubusercontent.com/assets/3418971/14922089/a0f7ecc2-0e03-11e6-91f8-073c462c7e8c.png)

The plugin's configuration panel looks like this:

![screenshot_2016-04-29_12-10-14](https://cloud.githubusercontent.com/assets/3418971/14922093/ae998048-0e03-11e6-814f-24f6b7c6ceb6.png)

Here's a partial picture of my desktop with a Pidgin conversation window open but unfocused:

![screenshot_2016-04-29_12-10-50](https://cloud.githubusercontent.com/assets/3418971/14922108/bdfd875a-0e03-11e6-8ade-7219925815bf.png)

If this person sends me a message, then the plugin sets the urgent hint on the conversation window, causing the following to happen:

![screenshot_2016-04-29_12-11-18](https://cloud.githubusercontent.com/assets/3418971/14922119/cc0f2b6e-0e03-11e6-9f53-01226f5541f7.png)

The specific behavior of "urgent" is dependent upon desktop environment.  The urgent hint is an X11-specific behavior.  That said, many windowing environments (e.g. Windows) have some notion similar to this, so there's probably some cross-platform way to set this behavior within webkit or something.  (I'm completely unfamiliar with developing in this area, so I'm hazarding a guess.)

If I then click on the button or otherwise give the window focus, the plugin removes the hint.  (See the plugin configuration panel above.)

There are a few blog posts like [this one](http://notes.secretsauce.net/notes/2014/03/19_x11-urgency-hint-and-notifications.html) that discuss the concept.

Is that what you were looking for?

**Comment by jnugh** [comment]
I just had a look at the electron docs, https://github.com/electron/electron/blob/master/docs/api/browser-window.md#winflashframeflag `win.flashFrame(true)` does the trick (at least on Unity).

It uses the the following code in chromium, looks like it should work in gtk environments in general.

```C
void BrowserWindowGtk::FlashFrame(bool flash) {
  // May not be respected by all window managers.
  gtk_window_set_urgency_hint(window_, flash);
}
```

I will prepare a PR for this, seems to be fairly easy as the feature is already used for windows :construction_worker:.

**Comment by yuya-oc** [comment]
Should be closed by #200 and #221
